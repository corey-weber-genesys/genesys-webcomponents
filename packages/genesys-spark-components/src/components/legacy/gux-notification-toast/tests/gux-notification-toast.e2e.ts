import { E2EPage } from '@stencil/core/testing';
import { newSparkE2EPage, a11yCheck } from '../../../../test/e2eTestUtils';

async function clickDismissButton(page: E2EPage) {
  await page.evaluate(async () => {
    const element = document.querySelector('gux-notification-toast-legacy');
    const dismissButtonElement =
      element.shadowRoot.querySelector('gux-dismiss-button');
    const button = dismissButtonElement.shadowRoot.querySelector('button');

    button.click();
  });
}
describe('gux-notification-toast-legacy', () => {
  describe('#render', () => {
    [
      {
        description: 'should render neutral notification toast',
        html: `
          <gux-notification-toast-legacy lang="en" accent="neutral">
            <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
            <div slot="title">Title</div>
            <div slot="message">This is the message</div>
          </gux-notification-toast-legacy>
        `
      },
      {
        description: 'should render positive notification toast',
        html: `
          <gux-notification-toast-legacy lang="en" accent="positive">
            <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
            <div slot="title">Title</div>
            <div slot="message">This is the message</div>
          </gux-notification-toast-legacy>
        `
      },
      {
        description: 'should render alert notification toast',
        html: `
          <gux-notification-toast-legacy lang="en" accent="alert">
            <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
            <div slot="title">Title</div>
            <div slot="message">This is the message</div>
          </gux-notification-toast-legacy>
        `
      },
      {
        description: 'should render warning notification toast',
        html: `
          <gux-notification-toast-legacy lang="en" accent="warning">
            <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
            <div slot="title">Title</div>
            <div slot="message">This is the message</div>
          </gux-notification-toast-legacy>
        `
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-notification-toast-legacy');
        await a11yCheck(page);
        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });

  describe('dismiss', () => {
    it('click dismiss button', async () => {
      const html = `
        <gux-notification-toast-legacy lang="en" accent="neutral">
          <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
          <div slot="title">Title</div>
          <div slot="message">This is the message</div>
        </gux-notification-toast-legacy>
      `;
      const page = await newSparkE2EPage({ html });
      const guxdismissSpy = await page.spyOnEvent('guxdismiss');
      const clickSpy = await page.spyOnEvent('click');

      expect(guxdismissSpy).not.toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-notification-toast-legacy')).not.toBeNull();

      await clickDismissButton(page);
      await page.waitForChanges();

      expect(guxdismissSpy).toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-notification-toast-legacy')).toBeNull();
    });

    it('click dismiss button and prevent default', async () => {
      const html = `
        <gux-notification-toast-legacy lang="en" accent="neutral">
          <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
          <div slot="title">Title</div>
          <div slot="message">This is the message</div>
        </gux-notification-toast-legacy>
      `;
      const page = await newSparkE2EPage({ html });
      const guxdismissSpy = await page.spyOnEvent('guxdismiss');
      const clickSpy = await page.spyOnEvent('click');

      await page.evaluate(() => {
        document.addEventListener('guxdismiss', event => {
          event.preventDefault();
        });
      });

      expect(guxdismissSpy).not.toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-notification-toast-legacy')).not.toBeNull();

      await clickDismissButton(page);
      await page.waitForChanges();

      expect(guxdismissSpy).toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-notification-toast-legacy')).not.toBeNull();
    });
  });
});
