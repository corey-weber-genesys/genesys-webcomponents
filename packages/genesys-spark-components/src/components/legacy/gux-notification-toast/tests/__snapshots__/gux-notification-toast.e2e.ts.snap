// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-notification-toast #render should render alert notification toast 1`] = `"<gux-notification-toast lang=\\"en\\" accent=\\"alert\\" hydrated=\\"\\"> <gux-icon slot=\\"icon\\" icon-name=\\"user-add\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon> <div slot=\\"title\\">Title</div> <div slot=\\"message\\">This is the message</div> </gux-notification-toast>"`;

exports[`gux-notification-toast #render should render neutral notification toast 1`] = `"<gux-notification-toast lang=\\"en\\" accent=\\"neutral\\" hydrated=\\"\\"> <gux-icon slot=\\"icon\\" icon-name=\\"user-add\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon> <div slot=\\"title\\">Title</div> <div slot=\\"message\\">This is the message</div> </gux-notification-toast>"`;

exports[`gux-notification-toast #render should render positive notification toast 1`] = `"<gux-notification-toast lang=\\"en\\" accent=\\"positive\\" hydrated=\\"\\"> <gux-icon slot=\\"icon\\" icon-name=\\"user-add\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon> <div slot=\\"title\\">Title</div> <div slot=\\"message\\">This is the message</div> </gux-notification-toast>"`;

exports[`gux-notification-toast #render should render warning notification toast 1`] = `"<gux-notification-toast lang=\\"en\\" accent=\\"warning\\" hydrated=\\"\\"> <gux-icon slot=\\"icon\\" icon-name=\\"user-add\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon> <div slot=\\"title\\">Title</div> <div slot=\\"message\\">This is the message</div> </gux-notification-toast>"`;
